File: JavaScript_Courses/JavaScript_Basics/Preparation/05_While_Loop_Lab/08_Graduation.js
function graduation(input) {
    // ако годишната му оценка е по-голяма или равна на 4.00. 
    // Ако ученикът бъде скъсан повече от един път, то той бива 
    // изключен и програмата приключва, като се отпечатва името на ученика и в кой клас бива изключен.
    let student = input[0];
    let fails = 0;
    let currentGrade = 1;
    let totalScore = 0;

    let i = 1;
    while (true) {
        if (currentGrade > 12) {
            console.log(`${student} graduated. Average grade: ${(totalScore/12).toFixed(2)}`);
            break;
        }

        let currentScore = Number(input[i]);
        i++;

        if (currentScore < 4) {
            fails += 1;
            if (fails == 2) {
                console.log(`${student} has been excluded at ${currentGrade} grade`);
                break;
            }
        } else {
            totalScore += currentScore;
            currentGrade++;
        }

    }
}


// graduation(["Gosho",
//     "5",
//     "5.5",
//     "6",
//     "5.43",
//     "5.5",
//     "6",
//     "5.55",
//     "5",
//     "6",
//     "6",
//     "5.43",
//     "5"]);

File: Python_Courses/Python_Basics/Python_PB_2024_06/05_While_Loop_Lab/08_Graduation.py
student = input()

grades_sum = 0
current_class = 1
fails_count = 0
is_graduated = True

while current_class <= 12:
    grade = float(input())
    if grade < 4:
        fails_count += 1
        if fails_count >= 2:
            is_graduated = False
            break
        continue

    grades_sum += grade
    current_class += 1

# is_graduated = fails_count < 2
if is_graduated:
    avg_grade = grades_sum / 12
    print(f'{student} graduated. Average grade: {avg_grade :.2f}')
else:
    print(f'{student} has been excluded at {current_class} grade')
